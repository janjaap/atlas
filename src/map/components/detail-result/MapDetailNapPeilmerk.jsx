import React from 'react';
import PropTypes from 'prop-types';

import MapDetailResultItem from './MapDetailResultItem';
import MapDetailResultWrapper from './MapDetailResultWrapper';

const MapDetailNapPeilmerk = ({ panoUrl, peilmerk, onMaximize, onPanoPreviewClick }) => (
  <MapDetailResultWrapper
    panoUrl={panoUrl}
    onMaximize={onMaximize}
    onPanoPreviewClick={onPanoPreviewClick}
    subTitle={peilmerk.label}
    title="NAP Peilmerk"
  >
    <ul className="map-detail-result__list">
      <MapDetailResultItem
        label="Hoogte NAP"
        value={peilmerk.height && `${peilmerk.height} m`}
      />
      <MapDetailResultItem
        label="Jaar"
        value={peilmerk.year && peilmerk.year.toString()}
      />
      <MapDetailResultItem
        label="Omschrijving"
        value={peilmerk.description}
      />
    </ul>
  </MapDetailResultWrapper>
);

MapDetailNapPeilmerk.propTypes = {
  panoUrl: PropTypes.string.isRequired,
  peilmerk: PropTypes.shape({
    description: PropTypes.string,
    height: PropTypes.string,
    label: PropTypes.string,
    year: PropTypes.number
  }).isRequired,
  onMaximize: PropTypes.func.isRequired,
  onPanoPreviewClick: PropTypes.func.isRequired
};

export default MapDetailNapPeilmerk;
