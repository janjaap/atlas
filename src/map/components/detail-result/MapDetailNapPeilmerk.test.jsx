import React from 'react';
import { shallow } from 'enzyme';

import MapDetailNapPeilmerk from './MapDetailNapPeilmerk';

describe('MapDetailNapPeilmerk', () => {
  it('should render everything', () => {
    const peilmerk = {
      description: 'Peilmerk description',
      height: 123.32313,
      label: 'Peilmerk label',
      year: 1919,
      windDirection: 'W',
      wallCoordinates: [2, 3]
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailNapPeilmerk
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        peilmerk={peilmerk}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render empty height', () => {
    const peilmerk = {
      description: 'Peilmerk description',
      height: null,
      label: 'Peilmerk label',
      year: 1919
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailNapPeilmerk
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        peilmerk={peilmerk}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render without height', () => {
    const peilmerk = {
      description: 'Peilmerk description',
      height: null,
      label: 'Peilmerk label',
      year: 1919
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailNapPeilmerk
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        peilmerk={peilmerk}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
