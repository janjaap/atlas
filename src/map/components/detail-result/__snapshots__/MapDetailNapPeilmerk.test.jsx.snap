// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailNapPeilmerk should not render empty height 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Peilmerk label"
  title="NAP Peilmerk"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultItem
      label="Hoogte NAP"
      value=""
    />
    <MapDetailResultItem
      label="Jaar"
      value="1919"
    />
    <MapDetailResultItem
      label="Omschrijving"
      value="Peilmerk description"
    />
  </ul>
</MapDetailResultWrapper>
`;

exports[`MapDetailNapPeilmerk should not render without height 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Peilmerk label"
  title="NAP Peilmerk"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultItem
      label="Hoogte NAP"
      value=""
    />
    <MapDetailResultItem
      label="Jaar"
      value="1919"
    />
    <MapDetailResultItem
      label="Omschrijving"
      value="Peilmerk description"
    />
  </ul>
</MapDetailResultWrapper>
`;

exports[`MapDetailNapPeilmerk should render everything 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Peilmerk label"
  title="NAP Peilmerk"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultItem
      label="Hoogte NAP"
      value="123 m"
    />
    <MapDetailResultItem
      label="Jaar"
      value="1919"
    />
    <MapDetailResultItem
      label="Omschrijving"
      value="Peilmerk description"
    />
  </ul>
</MapDetailResultWrapper>
`;
